doc/TockBinaryFormat.md

```diff
@@ -459,6 +520,82 @@ data between kernel and userspace and the the system call numbers.
+fields, `binary_end_offset` and `version`. The `binary_end_offset` field
+allows the kernel to identify where in the TBF object the application
+binary ends. The gap between the end of the application binary and
+the end of the TBF object can contain footers. 
+
+```
+0             2             4             6             8
++-------------+-------------+---------------------------+
+| Type (9)    | Length (20) | init_offset               |
++-------------+-------------+---------------------------+
+| protected_size            | min_ram_size              |
++---------------------------+---------------------------+
+| binary_end_offset         | version                   |
++---------------------------+---------------------------+
+```
+
+  * `init_offset` the offset in bytes from the beginning of binary payload
+    (i.e. the actual application binary) that contains the first instruction to
+   	execute (typically the `_start` symbol).
+  * `protected_size` the size of the protected region in bytes. Processes do not
+    have write access to the protected region. TBF headers are contained in the
+    protected region.
+  * `minimum_ram_size` the minimum amount of memory, in bytes, the process
+    needs.
+  * `binary_end_offset` specifies the offset from the beginning of the TBF
+    Object at which the Userspace Binary ends and optional footers begin.
+  * `version` specifies a version number for the application implemented by
+    the Userspace Binary. This allows a kernel to distinguish different 
+	versions of a given application.
+
+If a Program header is not present, `binary_end_offset` can be
+considered to be `total_size` of the Base Header and `version` is 0.
+
+The Main Header and Program Header have overlapping functionality. If
+a TBF Object has both, the kernel decides which to use.  Tock is
+transitioning to having the Program Header as the standard one to use,
+but older kernels (2.0 and earlier) do not recognize it and use the
+Main Header.
+
+#### `128` Credentials Footer
+
+A Credentials Footer contains cryptographic credentials for the integrity
+and possibly identity of a Userspace Binary. A Credentials Footer has
+the following format:
+
+```
+0             2             4             6             8
++-------------+-------------+---------------------------+
+| Type (128)  | Length (4+n)| format                    |
++-------------+-------------+---------------------------+
+| data...
++---------------------------+---------------------------+
+```
+
+The length of the data field is defined by the `Length` field. If
+the data field is `n` bytes long, the `Length` field is 4+n. The
+`format` field defines the format of the data field:
+
+```rust
+pub enum TbfFooterV2CredentialsType {
+    Reserved = 0,
+    CleartextID = 1,
+    Rsa3072Key = 2,
+    Rsa4096Key = 3,
+    Rsa3072KeyWithID = 4,
+    Rsa4096KeyWithID = 5,
+    SHA256 = 6,
+    SHA384 = 7,
+    SHA512 = 8,
+}
```

Comment:
Doesn't match TRD

Reply:
Sorry - forgot to push the change to this.